tox.ini

```diff
@@ -56,7 +56,10 @@ deps =
     wrapt
 
     ; Python 3.5+ only deps
-    py{37,38,39,310,311}: pytest-asyncio
+    py{37,38,39,310,311,312}: pytest-asyncio
+
+    ; For pkg_resources
+    py{37,38,39,310,311,312}: setuptools
```

Comment:
I am still not very clear on why this dependency is needed. You mentioned it was added since a test imports `pkg_resources`, which test are you referring to?

Reply:
Probably only “necessary” on Python 3.12 though:

> [gh-95299](https://github.com/python/cpython/issues/95299): Do not pre-install setuptools in virtual environments created with [venv](https://docs.python.org/3.12/library/venv.html#module-venv). This means that distutils, setuptools, pkg_resources, and easy_install will no longer available by default; to access these run pip install setuptools in the [activated](https://docs.python.org/3.12/library/venv.html#venv-explanation) virtual environment.

https://docs.python.org/3.12/whatsnew/3.12.html

Reply:
Perhaps, but it’s not harmful on others. The dependency is still real; it’s just allowed to be implicit. I guess I can change to `py312` only if implicit is better than explicit.